boruta/script.whizzml

```diff
@@ -0,0 +1,391 @@
+    (loop (f-rank init-rank fields ori-fields ds-id dataset-id runs 0)
+      ;; Repeats the procedure until the importance is assigned for all
+      ;; the attributes, or the algorithm has reached the previously set
+      ;; limit of the random forest runs.
+      (if (or (= runs max-runs) (full f-rank))
+          ;; as runs go by, the fields structure is modified and unimportant
+          ;; fields are removed. Thus we need f-names to contain the original
+          ;; list of names to know which are unimportant
+          (group-importance f-rank fields f-names objective)
+          ;; Loop body: Extends the information system by adding copies of all
+          ;; variables (the information system is always extended by at least
+          ;; 5 shadow attributes, even if the number of attributes in the
+          ;; original set is lower than 5)
+          ;; Shuffle the added attributes to remove their correlations with
+          ;; the response.
+          (let (sh-ds-info (shadow-ds ds-id fields objective)
+                sh-ds-id (head sh-ds-info)
+                sh-ids (head (tail sh-ds-info))
+                fields (nth sh-ds-info 2)
+                to-names (lambda (x) (get-in fields [x "name"]))
+                ;; Run a random forest classifier on the extended
+                ;; information system and gather the importances
+                rf-id (random-forest sh-ds-id objective)
+                ;; Find the maximum importance among shadow attributes,
+                ;; and then assign a hit to every attribute that scored better
+                ;; by more than the min-gain margin or worst by less than the
+                ;; min-gain margin.
+                imps (importances rf-id)
+                _ (log-info "Number of runs: " (+ runs 1))
+                _ (log-info "------------------------------------------------")
+                max-sh (maximum-shadow-importance imps sh-ids)
+                _ (log-info "Max importance for the shadow fields: " max-sh)
+                f-rank (rank-field-ids imps sh-ids max-sh min-gain)
+                _ (log-info "Important: " (select-by-importance f-rank
+                                                                IMPORTANCE_INC
+                                                                to-names))
+                _ (log-info "Unimportant: " (select-by-importance f-rank
+                                                                  IMPORTANCE_DEC
+                                                                  to-names))
+                _ (log-info "Neutral: " (select-by-importance f-rank
+                                                              0
+                                                              to-names))
+                fields (remove-unimportant fields f-rank)
+                _ (log-info "remaining fields: " (map to-names (keys fields)))
+                _ (log-info "------------------------------------------------"))
```

Comment:
you can put all these logs in the body of the `let`, right before he `recur`, without having to assign them to _ and with a bit more space :)